### DISCUSSION TITLE
64bit math support  in Berry

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#19639] [General]
Is there 64-bit math support in Tasmota Berry?? if so, is there any documentation or location of the source code library...

thanks

**Comment by s-hadinger** [comment]
Yes, there is limited support with `int64` class, from here: https://github.com/arendst/Tasmota/tree/development/lib/libesp32/berry_int64

Buy it's not documented yet.

Definition can give you some ideas about what's supported:
```
  set, ctype_func(int64_set)
  fromu32, static_ctype_func(int64_fromu32)

  tostring, ctype_func(int64_tostring)
  fromstring, static_ctype_func(int64_fromstring)
  toint, ctype_func(int64_toint)

  +, ctype_func(int64_add)
  -, ctype_func(int64_sub)
  *, ctype_func(int64_mul)
  /, ctype_func(int64_div)
  %, ctype_func(int64_mod)
  -*, ctype_func(int64_neg)
  ==, ctype_func(int64_equals)
  !=, ctype_func(int64_nequals)
  >, ctype_func(int64_gt)
  >=, ctype_func(int64_gte)
  <, ctype_func(int64_lt)
  <=, ctype_func(int64_lte)

  tobytes, ctype_func(int64_tobytes)
  frombytes, ctype_func(int64_frombytes)
```

As well as unit tests:
https://github.com/arendst/Tasmota/blob/development/lib/libesp32/berry_int64/tests/int64.be

**Comment by sfromis** [comment]
If you instead mean double-precision 64-bit floating point math, config setting `BE_USE_SINGLE_FLOAT` suggests a selection between float/double as type, but IDK how much this is implemented.
